testing/src/test/java/io/stargate/it/http/graphql/graphfirst/FilesResourceTest.java

```diff
@@ -0,0 +1,191 @@
+    // given
+    OkHttpClient httpClient = getHttpClient();
+    String url = String.format("%s:8080/graphqlv2/files/cql_directives.graphql", host);
+    Request getRequest =
+        new Request.Builder()
+            .get()
+            .addHeader("content-type", MediaType.TEXT_PLAIN)
+            .url(url)
+            .build();
+
+    // when
+    Response response = httpClient.newCall(getRequest).execute();
+
+    // then
+    assertThat(response.code()).isEqualTo(200);
+    assertThat(response.body()).isNotNull();
+    String responseBody = response.body().string();
+    assertThat(responseBody).isNotEmpty();
+    assertThat(responseBody).contains("directive @cql_input");
+  }
+
+  @Test
+  public void shouldGetFileWithSchema(@TestKeyspace CqlIdentifier keyspaceId) throws IOException {
+    // when get schema file
+    String url = String.format("%s:8080/graphqlv2/files/namespace/%s.graphql", host, keyspaceId);
+    Request getRequest =
+        new Request.Builder()
+            .get()
+            .addHeader("content-type", MediaType.TEXT_PLAIN)
+            .url(url)
+            .build();
+
+    Response getSchemaResponse = getHttpClient().newCall(getRequest).execute();
+
+    // then
+    assertThat(getSchemaResponse.code()).isEqualTo(200);
+    assertThat(getSchemaResponse.header("Content-Disposition"))
+        .contains((String.format("%s-%s.graphql", keyspaceId.toString(), DEPLOYED_SCHEMA_VERSION)));
+
+    String responseBody = getSchemaResponse.body().string();
+    assertThat(responseBody).isNotEmpty();
+    assertThat(responseBody).contains("type User");
+    assertThat(responseBody).contains("type Query");
+  }
+
+  @Test
+  public void shouldGetFileWithSchemaUsingVersionArgument(@TestKeyspace CqlIdentifier keyspaceId)
+      throws IOException {
+    // when get schema file for the deployedVersion
+    String url =
+        String.format(
+            "%s:8080/graphqlv2/files/namespace/%s.graphql?version=%s",
+            host, keyspaceId, DEPLOYED_SCHEMA_VERSION);
+    Request getRequest =
+        new Request.Builder()
+            .get()
+            .addHeader("content-type", MediaType.TEXT_PLAIN)
+            .url(url)
+            .build();
+
+    Response getSchemaResponse = getHttpClient().newCall(getRequest).execute();
+
+    // then
+    assertThat(getSchemaResponse.code()).isEqualTo(200);
+    assertThat(getSchemaResponse.header("Content-Disposition"))
+        .contains((String.format("%s-%s.graphql", keyspaceId.toString(), DEPLOYED_SCHEMA_VERSION)));
+
+    String responseBody = getSchemaResponse.body().string();
+    assertThat(responseBody).isNotEmpty();
+    assertThat(responseBody).contains("type User");
+    assertThat(responseBody).contains("type Query");
+  }
+
+  @Test
+  public void shouldReturnNotFoundIfSchemaDoesNotExists() throws IOException {
+    // when get schema file
+    String url = String.format("%s:8080/graphqlv2/files/namespace/%s.graphql", host, "unknown");
```

Comment:
What about correct namespace and wrong version? Is that a 404 or a 400?

Reply:
It will be a 404 as well. I'll add a test.